li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/producer/LiKafkaProducerConfig.java

```diff
@@ -98,6 +99,8 @@
       "and thus not split into multiple messages. This configuration does not have any effect if large message is " +
       "not enabled.";
 
+  public static final String ENABLE_RECORD_HEADER_DOC = "If true, we will use record header based large message support and stop using segmentSerializer.";
```

Comment:
Is this config only used for large message support?  If so, could you rename the config to include "large.message" in the config name?  If not, could you update the doc to mention there could be other use of it

Reply:
It will be used for large message support and other kinds of serialization(e.g. encryption or compression). I have updated the comments